app/src/main/java/org/wikipedia/recommendedcontent/RecommendedContentFragment.kt

```diff
@@ -119,12 +122,16 @@ class RecommendedContentFragment : Fragment() {
     }
 
     private fun buildRecommendedContent(list: List<PageSummary>) {
-        (requireParentFragment().requireParentFragment() as SearchFragment).onSearchProgressBar(false)
+        parentSearchFragment.onSearchProgressBar(false)
         if (list.isEmpty()) {
+            parentSearchFragment.analyticsEvent = ExperimentalLinkPreviewInteraction(HistoryEntry.SOURCE_SEARCH, RecommendedContentAnalyticsHelper.abcTest.getGroupName(), false)
```

Comment:
Not sure but do we need to send group-specific source here?

Reply:
I don't believe so. I think for the `impression` event the source is not relevant.